web-portal/backend/src/tenant/tenant.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable, Inject } from '@nestjs/common';
+import { CustomPrismaService } from 'nestjs-prisma';
+import { PrismaClient, TransactionType } from '../../../../.generated/client';
+import { randomBytes } from 'crypto';
+import * as bcrypt from 'bcrypt';
+
+@Injectable()
+export class TenantService {
+  constructor(
+    @Inject('Postgres')
+    private prisma: CustomPrismaService<PrismaClient>, // <-- Inject the PrismaClient
+  ) {}
+
+  async countAll(): Promise<number> {
+    const count = await this.prisma.client.tenant.count();
+    return count;
+  }
+
+  async create(): Promise<any> {
+    const secretKey = randomBytes(8).toString('hex');
+    const salt = await bcrypt.genSalt();
+
+    const hashedKey = await bcrypt.hash(secretKey, salt);
```

Comment:
I don't think this works for our usecase. We want the tenant secret to be both the lookup key and hashed to secure it. This method (as seen below) requires a separate lookup key so the comparison only has to be done once. Is there a way to either use a shared salt so the hash is deterministic?